[EP-399] QA Phase 1 Remove Properties

moderateepheezy opened the pull request:
<!-- This template is **just a guide**, delete any and all parts which you don't need! -->

# 📲 What

Remove the following event properties and their related tests.

- checkout_amount
- checkout_bonus_amount
- checkout_shipping_amount
- context_timestamp
- project_category_id
- project_goal
- project_location
- project_static_usd_rate
- project_subcategory_id
- session_apple_pay_device
- session_cellular_connection
- session_enabled_features
- session_mp_lib
- session_referrer_credit
- session_wifi_connection
- user_country



# 🤔 Why

Base on the feedback from QA for Phase 1 event tracking, some of the event  properties sent to Segment are not needed.

singhhari reviewed (changes requested):
I know most of this consists of deletions but is there a way to bump the coverage a bit?

moderateepheezy commented:
> I know most of this consists of deletions but is there a way to bump the coverage a bit?

I have tried, unless I go out of scope of my task, not sure if there's any easier way to increase the coverage. I also checked some functions that I didn't find a direct test for in `SharedFunction`, and I added tests for them, but that didn't increase as well as something else already provided coverage for them, so I deleted them.

moderateepheezy commented on Library/ViewModels/ProjectPamphletViewModel.swift:
```diff
@@ -186,12 +186,11 @@ public final class ProjectPamphletViewModel: ProjectPamphletViewModelType, Proje
 
     freshProjectRefTagAndCookieRefTag
-      .observeValues { project, refTag, cookieRefTag in
+      .observeValues { project, refTag, _ in
```
`freshProjectRefTagAndCookieRefTag `is used elsewhere in the code with `cookieRefTag` needed.

singhhari replied:
i just did a global search on `freshProjectRegTagAndCookieTag` and don't see it used anywhere outside this file. I think you can remove the third type parameter from line 177, update the variable name and the observeValues block to remove the `_`.

moderateepheezy commented on Library/Tracking/KSRAnalytics.swift:
```diff
@@ -1025,8 +1021,7 @@ public final class KSRAnalytics {
     pledgeViewContext: PledgeViewContext,
     checkoutData: CheckoutPropertiesData,
-    refTag: RefTag?,
-    cookieRefTag: RefTag?
```
`cookieRefTag` is used for `referrerCredit` which is one of the properties I'm removing.

moderateepheezy commented on Library/Tracking/KSRAnalytics.swift:
```diff
@@ -1358,13 +1352,11 @@ public final class KSRAnalytics {
     _ project: Project,
     refTag: RefTag? = nil,
-    sectionContext: KSRAnalytics.SectionContext,
-    cookieRefTag: RefTag? = nil
```
`cookieRefTag` is used for `referrerCredit` which is one of the properties I'm removing.

moderateepheezy commented on Library/Tracking/KSRAnalytics.swift:
```diff
@@ -1676,12 +1649,10 @@ private func checkoutProperties(from data: KSRAnalytics.CheckoutPropertiesData,
   -> [String: Any] {
   var result: [String: Any] = [:]
 
-  result["amount"] = data.amount
```
No they are not, what I removed is `checkout_amount`

singhhari commented on Library/ViewModels/PledgeViewModel.swift:
```diff
@@ -849,8 +849,7 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
           checkoutData: checkoutData,
-          refTag: refTag,
-          cookieRefTag: cookieRefTag
+          refTag: refTag
```
line 831 has a constant that is no longer being used. Delete that.

singhhari reviewed (approved):
lgtm
